ddtrace/contrib/starlette/patch.py

```diff
@@ -65,3 +58,36 @@ def unpatch():
     setattr(starlette, "_datadog_patch", False)
 
     _u(starlette.applications.Starlette, "__init__")
+
+    handlers = [
+        starlette.routing.Mount,
+        starlette.routing.Route,
+        starlette.routing.Host,
+        starlette.routing.WebSocketRoute,
+    ]
+
+    # Unwrap all of the handlers
+    for handler in handlers:
+        _u(handler, "handle")
+
+
+def traced_handler(wrapped, instance, args, kwargs):
+    def _wrap(scope, receive, send):
+        # Since handle can be called multiple times for one request, we take the path of each instance
+        # Then combine them at the end to get the correct resource name
+        if isinstance(instance, BaseRoute):
+            if "__dd_paths__" in scope:
+                scope["__dd_paths__"].append(instance.path)
+
+            else:
+                scope["__dd_paths__"] = [instance.path]
```

Comment:
We should already have a `scope["datadog"]` from the [ASGI middleware](https://github.com/DataDog/dd-trace-py/blob/7919523fc/ddtrace/contrib/asgi/middleware.py#L122-L122) which is a dict.

So as long as that code (always) runs before this code, then we can likely just do:

```python
# is "route_paths" a good name?
if "route_paths" in scope["datadog"]:
    scope["datadog"]["route_paths"].append(instance.path)
else:
    scope["datadog"]["route_paths"] = [instance.path]
```

We could also do:

```python
scope["datadog"].setdefault("route_paths", [])
scope["datadog"]["route_paths"].append(instance.paths)
```

Reply:
Yeah I tried to do that but it didn't seem to work. When looking at it, it seemed `scope["datadog"]` is created on a per span basis, so it gets reset with each new span or each new handler hit, therefore it won't work to store the `route_paths` on. 

If that sounds wrong to you though I'll go back and look it more in depth.

Reply:
🤔 hmm, I would think it would be available.

If it isn't there ever, then sure, what you have makes sense, but if `datadog` is consistently there, I'd prefer to store in there

Reply:
Tested again, the sub app test fails when we put it in the datadog scope dict.

Reply:
We discussed this off of GitHub, the ASGIMiddleware/setting `scope["datadog"] = {}` happens _after_ this code is called.

We can leave this as-is for now.